src/seer/automation/autofix/event_manager.py

```diff
@@ -130,7 +130,10 @@ def set_selected_root_cause(self, payload: AutofixRootCauseUpdatePayload):
 
     def send_coding_start(self):
         with self.state.update() as cur:
-            plan_step = cur.add_step(self.plan_step)
+            plan_step = cur.find_or_add(self.plan_step)
```

Comment:
```suggestion
            plan_step = cur.find_step(key=self.plan_step.key)
```
Would it be cleaner to do explicit finding then adding only if it doesn't exist here?

Reply:
I did that to use the same exact logic as the root cause step, but I've updated both now because you're right that is cleaner.